Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs

```diff
@@ -1052,16 +1052,9 @@ internal void UpdateLeftBarButtonItem(Page pageBeingRemoved = null)
 
 				var currentChild = this.Child;
 				var firstPage = n.NavPage.Pages.FirstOrDefault();
-
-
+				
 				if (n._parentMasterDetailPage == null)
-					return;
-
-				if (firstPage != pageBeingRemoved && currentChild != firstPage && NavigationPage.GetHasBackButton(currentChild))
```

Comment:
Check this comment https://github.com/xamarin/Xamarin.Forms/pull/3939/files#r221403674